packages/run-protocol/src/reserve/assetReserve.js

```diff
@@ -246,11 +247,11 @@ const start = async (zcf, privateArgs) => {
     const collateralKeyword = getKeywordForBrand(collateralAmount.brand);
     if (
       !AmountMath.isGTE(
-        collateralSeat.getCurrentAllocation()[collateralKeyword],
+        collateralSeat.getCurrentAllocation()[+collateralKeyword],
```

Comment:
`+` coerces to a number. I don't think this can work. Does it pass tests?

I suggest an `extractFieldFromKeywordRecord(record, key)` library function imported from a module with keyword record helpers. Perhaps with a more concise name. Or perhaps there's some `Object.getProperty` standard JS function that would work?

Reply:
The helper function would not be in a `@jessie-check` file. Just like `makeSet` and such.

> what should it say instead?

Maybe "if you're trying to do numeric indexing, prefix the index with a +". I agree that diagnostic is misleading.

Reply:
> The helper function would not be in a @jessie-check file. Just like makeSet and such.

Sure. But won't it still run in a Jessie environment? What's the point of putting it in a different module? I thought jessie-check was for ensuring that something can safely run in a Jessie environment. Some of these rules sound paternalistic, like "you shouldn't use curlies because you might misread them". Is the arbitrarily computed property rule like that or is it a runtime constraint?

Reply:
Yes, the primary purpose of Jessie, and the only purpose for which we are currently using it, is to help us write more reliable and analyzable code. To be fully pedantic, the `// @jessie-check` enforces only Tiny-SES rather than Jessie, where the difference is the primordials the code is allowed to assume exists. Aside from the primordials the code can assume exists, Jessie and Tiny-SES are identical.

In practice that was one too many distinctions so we use Jessie to describe both. In practice, when we say we're writing most of our code in Jessie, we do not generally mean to imply that the code cannot assume the presence of, for example `Reflect`, which is part of SES but not Jessie. 

We eventually intend to support Jessie as a std for universal mobile code, where a Jessie interpreter can be implemented simply in any normal imperative language. For that purpose, this difference will matter, as that implementation can only be simple if it omits many of the SES primordials including `Reflect`.

Reply:
The purpose of the square bracket indexing rules is to make code that is using computed property lookup to index into arrays not be able to be tricked into looking up, for example, inherited method properties. In general, we want code that is using `[]` safely to index into a collection of some sort safely to be analyzably safe. Zoe introduced the notion of keywords for exactly the same reason, noting that at least all the primordial method names begin with a lowercase character, and so cannot be mistaken for keywords.

If we were willing to have Jessie understand that keywords are safe computed indexes, we could similarly enable syntax that enforced that the index expression evaluated to a keyword. We have not bothered to do that, and have instead just tried to use `[]` safely by feeding it only keywords, which a static or dynamic check ensuring this.

So we have choices wrt these uses of `record[keyword]` in what would otherwise be Jessie code:
   * Have Jessie drop the `[]` restrictions. Yes, we should consider this.
   * Rewrite such code to `Reflect.get(record, keyword)` which means the same thing *given* the availability of `Reflect` which is technically part of SES and Tiny-SES but not Jessie.
   * Introduce a keyword enforcing helper function, upgrade Jessie to allow it, and use it for keyword lookup in the same way we use `+` for array lookup.
   * Do we have a `// @jessie-expect-error` directive analogous to `// @ts-expect-error`? If so, you could use it to allow expressions you know to be safe -- perhaps even adding to the comment directive an explanation of why you believe it to be safe.

The path of least resistance for now is probably rewriting to `Reflect.get(record, keyword)`. That's what I'd do, but feel free to consider the other alternatives.